### PR TITLE
Add relatedResource tests.

### PR DESCRIPTION
**Pull request by BigBlueHat**
- Implement relatedResource initial test.
- Add relatedResource missing id test.
- Add relatedResource digest required test.
- Update relatedResource test to test mediaType.

### CODE REVIEW

File: tests/input/relatedResource/relatedResource-missing-id-fail.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  ],
+  "type": [
+    "VerifiableCredential"
+  ],
+  "credentialSubject": {
+    "id": "did:example:subject"
+  },
+  "relatedResource": [{
+    "url": "https://www.w3.org/ns/credentials/v2",
+    "digestMultibase": "uEres1usWcWCmW7uolIW2uA0CjQ8iRV14eGaZStJL73Vz"
```
Comment (aljones15) [inline tests/input/relatedResource/relatedResource-missing-id-fail.json:13]: how did you produce digestMultibase here?

**Review by aljones15 — APPROVED** [review]
LGTM do want to know how to produce digestMultibase though

File: tests/50-advanced-concepts.js [hunk 2]
```diff
@@ -53,24 +53,38 @@ describe('Advanced Concepts', function() {
...
+        )), 'Failed to accept a VC with valid relatedResource objects.');
+        await assert.doesNotReject(endpoints.issue(require(
+          './input/relatedResource/relatedResource-with-mediaType-ok.json'
+        )),
+        'Failed to accept a VC with valid relatedResource.mediaType values.');
+        await assert.rejects(endpoints.issue(require(
+          './input/relatedResource/relatedResource-list-of-strings-fail.json'
+        )),
+        {name: 'HTTPError'},
+        'Failed to reject VC with relatedResource as an array of strings.');
```
Comment (TallTed) [inline tests/50-advanced-concepts.js:68]:
```suggestion
        'Failed to reject a VC with a relatedResource as an array of strings.');
```

File: tests/50-advanced-concepts.js [hunk 3]
```diff
@@ -53,24 +53,38 @@ describe('Advanced Concepts', function() {
...
         'among the list of related resource objects.', async function() {
         this.test.link = `https://w3c.github.io/vc-data-model/#integrity-of-related-resources:~:text=The%20identifier%20for%20the%20resource%20is%20REQUIRED%20and%20conforms%20to%20the%20format%20defined%20in%20Section%204.4%20Identifiers.%20The%20value%20MUST%20be%20unique%20among%20the%20list%20of%20related%20resource%20objects.`;
-        // TODO: implement
-        this.test.skipMessage = 'TBD';
-        this.skip();
+        await assert.rejects(endpoints.issue(require(
+          './input/relatedResource/relatedResource-missing-id-fail.json'
+        )),
+        {name: 'HTTPError'},
+        'Failed to reject VC with relatedResource missing its `id` field.');
```
Comment (TallTed) [inline tests/50-advanced-concepts.js:78]:
```suggestion
        'Failed to reject a VC with a relatedResource with no `id` field.');
```

File: tests/50-advanced-concepts.js [hunk 4]
```diff
@@ -53,24 +53,38 @@ describe('Advanced Concepts', function() {
...
         'a digestSRI or a digestMultibase value.', async function() {
         this.test.link = `https://w3c.github.io/vc-data-model/#integrity-of-related-resources:~:text=Each%20object%20associated%20with%20relatedResource%20MUST%20contain%20at%20least%20a%20digestSRI%20or%20a%20digestMultibase%20value.`;
-        // TODO: implement
-        this.test.skipMessage = 'TBD';
-        this.skip();
+        await assert.rejects(endpoints.issue(require(
+          './input/relatedResource/relatedResource-no-digest-fail.json'
+        )),
+        {name: 'HTTPError'},
+        'Failed to reject VC with relatedResource missing digest info.');
```
Comment (TallTed) [inline tests/50-advanced-concepts.js:87]:
```suggestion
        'Failed to reject a VC with a relatedResource with no digest info.');
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Tiny, human-facing

**Review by PatStLouis — APPROVED** [review]

File: tests/input/relatedResource/relatedResource-missing-id-fail.json (hunk 1 above)
Comment (BigBlueHat) [reply tests/input/relatedResource/relatedResource-missing-id-fail.json:13]: @aljones15 I copied it from this example in the spec: https://w3c.github.io/vc-data-model/#example-usage-of-the-relatedresource-and-digestmultibase-property

It is likely wrong at this point...

I could pick a different resource served by this test suite to use instead. Sound better?

Comment (BigBlueHat) [reply tests/input/relatedResource/relatedResource-missing-id-fail.json:13]: If we go the route of including related resources in this repo for these tests, we will have to:
1. serve them with the report
2. prevent them from changing without updating the hash (probably via a README warning)
3. document how their hashes were created for anyone who wants to change them later

**Comment by BigBlueHat** [comment]
@PatStLouis @aljones15 I added a locally served JSON file (an empty object, fwiw) for testing `relatedResource` values. This should minimize concerns about remote resources changing over time.

**Review by TallTed — APPROVED** [review]

**Pull request merged**